modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/LogicalRelImplementor.java

```diff
@@ -1104,4 +1123,38 @@ private static BinaryTupleSchema fromTableDescriptor(TableDescriptor descriptor)
 
         return BinaryTupleSchema.create(elements);
     }
+
+    private static boolean canFuseProjectionInto(RelNode rel) {
+        if (rel instanceof Join) {
+            Join join = (Join) rel;
+
+            return joinNeedsProjection(join);
+        }
+
+        return false;
+    }
+
+    private static boolean joinNeedsProjection(Join join) {
+        return join.getJoinType() == JoinRelType.INNER
+                || join.getJoinType() == JoinRelType.LEFT
+                || join.getJoinType() == JoinRelType.FULL
+                || join.getJoinType() == JoinRelType.RIGHT;
+    }
```

Comment:
This can be replaced with EnumSet
```suggestion
    EnumSet joinNeedsProjection = EnumSet.of(JoinRelType.INNER, JoinRelType.LEFT, JoinRelType.FULL, JoinRelType.RIGHT);
   // replace old method call with  `joinNeedsProjection.contains(join)`
```